### PR TITLE
chore: adds circleCI

### PR DESCRIPTION
**Pull request by chloelbn**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no     
| Need Doc update   | no

## Describe your change

Switch from Travis to CircleCI

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -8,8 +8,8 @@ if defined?(RUBY_ENGINE) && RUBY_ENGINE == 'rbx'
 end
 
 group :test do
-  rails_version = ENV["RAILS_VERSION"] ? "~> #{ENV["RAILS_VERSION"]}" : '>= 3.2.0'
-  gem 'rails', rails_version
+  rails_version = ENV["RAILS_VERSION"] ? "#{ENV["RAILS_VERSION"]}" : '5.2'
```
Comment (DevinCodes) [inline Gemfile:11]:
```suggestion
  rails_version = ENV["RAILS_VERSION"] || '5.2'
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    jobs:
+      - tests_rails_4:
+          matrix:
+            parameters:
+              version: ['2.2', '2.3', '2.4']
+              rails-version: ['4.2']
+              sequel-version: ['4.0']
+          filters:
+            tags:
+              only: /.*/
```
Comment (DevinCodes) [inline .circleci/config.yml:136]: What does this do, exactly?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      BUNDLE_RETRY: 3
+      BUNDLE_PATH: vendor/bundle
+      RAILS_VERSION: << parameters.rails-version >>
+      SEQUEL_VERSION: << parameters.sequel-version >>
+
+version: 2.1
+
+jobs:
+  tests_rails_3:
+    description: Build, unit and integration tests for Ruby
```
Comment (DevinCodes) [inline .circleci/config.yml:48]: Can you remove this for the other job descriptions too, please?
```suggestion
    description: Build, unit and integration tests
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,154 @@
+aliases:
+  - &credentials
+    name: Retrieve temporary Algolia credentials if needed
+    command: |
+      if [ -z "ALGOLIA_ADMIN_KEY_MCM" ]; then
+        curl -s https://algoliasearch-client-keygen.herokuapp.com | sh >> $BASH_ENV
+      fi
```
Comment (DevinCodes) [inline .circleci/config.yml:7]: Maybe we want to move this to just before we run the tests? This way, any errors with bundler surface quicker, and we don't download the keys unless we're actually going to run the tests, if that makes sense 🙂

**Review by DevinCodes — DISMISSED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (chloelbn) [reply .circleci/config.yml]: it should be deleted, it triggers your job on specific tags

**Comment by DevinCodes** [comment]
Do you know why the tests are failing?

**Comment by chloelbn** [comment]
@DevinCodes it's fixed now!

**Review by DevinCodes — APPROVED** [review]

**Pull request merged**